### PR TITLE
doc: cloud restructure

### PR DESCRIPTION
**Pull request by snkas**
Restructure the cloud documentation into its own directory with separate files and add extra explanation/wording.

Is this a user-visible change (yes/no): yes

### CODE REVIEW

**Review by lalithsuresh — APPROVED** [review]

File: docs/cloud/assets.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Assets
+
+In its installation, Feldera Cloud requires several assets, most notably
+Helm packages and container images. These assets are made available to you
+along with your (trial) license key. They are not publicly available.
```
Comment (mohsinbeg) [inline docs/cloud/assets.md:5]: > along with your (trial) license key. They are not publicly available.

along with your Feldera license key. They are not publicly available.

File: docs/cloud/intro.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+A portion of the Feldera assets (such as container images) required to
+actually perform the steps in this documentation are not publicly
+available, and are only accessible with a (trial) license key.
+
+Please contact us at `learnmore@feldera.com` if you are interested to learn
+more or would like a trial. We'd be happy to get you started!
+
+:::tip
+Don't forget to first try out the [Docker demo container](../intro), which requires
+no (trial) license key. The Docker demo already showcases many of Feldera's features.
```
Comment (mohsinbeg) [inline docs/cloud/intro.md]: > no (trial) license key.

no license key.

File: docs/cloud/intro.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+This documentation is about setting up a distributed cloud deployment.
+:::
+
+## Setup
+
+1. **[Load assets](assets):** using your (trial) license key, load the required assets
```
Comment (mohsinbeg) [inline docs/cloud/intro.md]: > using your (trial) license key

using your license key.

File: docs/cloud/intro.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+   into your cloud environment such that they are available for deployment.
+
+2. **[Deployment](deployment):** start a Kubernetes cluster on which Feldera Cloud 
+   is deployed.
+
+## Usage
+
+Once you have completed the setup, you can get started with making use of all
```
Comment (mohsinbeg) [inline docs/cloud/intro.md]: > Once you have completed the setup, you can get started with making use of all that Feldera has to offer.

Once you have completed the setup, you can get started with using Feldera Cloud.

File: docs/cloud/deployment.md [hunk 5]
```diff
@@ -70,7 +70,7 @@ eksctl create cluster -f eks-config.yaml
 ```
 
 This uses CloudFormation behind-the-scenes to bring up an EKS cluster named
-`feldera-eks` with three EC2 on-demand instances as worker nodes, 
+`feldera-eks` with three EC2 on-demand instances as worker nodes,
```
[the remark this answers is not in the document]
Comment (snkas) [reply docs/cloud/deployment.md:73]: I also see no requirement of on-demand on the instances in `eks-config.yaml` @lalithsuresh ? If not relevant anymore, I'll just remove the "on-demand" adjective.

[the remark this answers is not in the document]
Comment (lalithsuresh) [reply docs/cloud/deployment.md:73]: The sentence is correct in that the specified eksctl config file will bring up on-demand instances and not spot or reserved instances. I think just removing "on-demand" should be fine.

File: docs/cloud/assets.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# Step 1: Assets
+
+In its installation, Feldera Cloud requires several assets, most notably
+Helm packages and container images. These assets are made available to you
+along with your Feldera license key. They are not publicly available.
+Please contact us at `learnmore@feldera.com` if you would like a trial.
```
Comment (snkas) [inline docs/cloud/assets.md:6]: Shall I fully get rid of "trial" mention, as in reword this to "(...) if you would like to try out Feldera Cloud." ?

File: docs/cloud/index.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+Feldera Cloud is the enterprise offering of Feldera. It is designed such that
+you can bring your own cloud, on which you deploy Feldera.
+The documentation we provide here is publicly available, both to guide
+our existing customers and to provide insight to interested parties.
+A portion of the Feldera assets (such as container images) required to
+actually perform the steps in this documentation are not publicly
+available, and are only accessible with a license key.
+
+Please contact us at `learnmore@feldera.com` if you are interested to learn
+more or would like a trial. We'd be happy to get you started!
```
Comment (snkas) [inline docs/cloud/index.md:12]: Shall I fully get rid of "trial" mention, as in reword this to "(...) or would like to try out Feldera Cloud." ?

Comment (snkas) [reply docs/cloud/index.md]: Changed to the above.

**Comment by snkas** [comment]
Removed the "trial" mention and the minor rewording. Added CHANGELOG mention. Shall I go ahead and merge?
